### ISSUE TITLE
Trainning detail about the first stage

### ISSUE THREAD

**Issue by darkbridge** [jkwang28/OSDFace#11]
Hi, this is really an exciting work. I have a quick question about the first-stage training strategy mentioned in the paper: it says the HQ and LQ dictionaries are trained separately for 50 and 10 epochs. Does this mean we first train the HQ dictionary alone using non-degraded high-quality data with the VQ loss, and then fix the HQ dictionary while training the LQ dictionary for another 10 epochs using HQ-degraded data pairs with total loss? Am I misunderstanding something here?

**Comment by gobunu** [comment]
Thanks for the question. Our stage-1 training is done in two parts.

We first train the VRE with the association term turned off by setting λ<sub>assoc</sub> to 0. In this part, the HQ dictionary is trained for 50 epochs, while the LQ dictionary is trained for 10 epochs. The reason is that the LQ side is usually easier to fit and converges much faster, so it does not need as many epochs as the HQ dictionary.

After these dictionaries are ready and stable, we keep training for an additional 10 epochs and set λ<sub>assoc</sub> to 1 to activate the association constraint. Enabling it later helps avoid unstable alignment early on and works better once both dictionaries have already learned meaningful code items.

**Comment by darkbridge** [comment]
Thank you for your response. I have an additional question: How can one determine whether the first-stage training effect is stable? Is it possible to directly assess the results by examining the encode-quantize-decoded images?

**Comment by gobunu** [comment]
Yes, stability in the first stage can be judged quite directly.

A simple and practical way is to look at the reconstruction quality of the encode–quantize–decode pathway. If the training is stable, the reconstructed images should be visually close to the inputs and the typical artifacts such as blocking, color shifts, or identity drift should be reduced and no longer change much over time.

In addition to visual inspection, we usually monitor quantitative signals, such as the reconstruction loss curve and its convergence. It is also common to evaluate on a validation set with metrics like PSNR and LPIPS. When both the loss and the validation metrics plateau and the reconstructions stop improving noticeably, it is a good indication that the first-stage training has become stable.

**Comment by darkbridge** [comment]
Thank you for your response. 
Based on the provided inference code, I attempted to define the decoder as the MultiHeadDecoder class and initialized it using the provided weights. After performing encoder-quantize-decoder on the LQ image, the resulting image shown below is not a normal RGB-domain image. Could there be an issue with this approach?

<img width="1847" height="987" alt="Image" src="https://github.com/user-attachments/assets/d9135994-80b9-42dd-8662-b14b8bbe18c3" />

**Comment by jkwang28** [comment]
Looks like some data range bugs?
